### PR TITLE
Add debug tracing for API calls and HTTP requests

### PR DESCRIPTION
**Pull request by rls-p4**
Also add file debug.R, which contains functions for tracing/logging API calls and HTTP requests.

To enable tracing at runtime, use the following options:
* `options(scidb.debug=TRUE)`: enable debug printing
* `options(scidb.trace=TRUE)`: enable trace printing
* `options(scidb.debug=2)`: enable trace printing (legacy compatibility)
* `options(scidb.trace.http=TRUE)`: enable trace printing for HTTP requests
* `options(scidb.trace.api=TRUE)`: trace API calls declared with `@export`
* `options(scidb.trace.api='internal')`: trace important internal calls too
* `options(scidb.log.mask=TRUE)`: mask UIDs and hashes in logging output to make it easier to compare logs between two different runs.

To add tracing to a function, add a call to `.TraceEnter` and `.TraceExit` near the top of the function, like this:

```
Foo <- function(arg1, arg2, ...) {
  trace <- .TraceEnter("Foo", arg1=arg1, arg2=arg2, ...)
  on.exit(.TraceExit(trace, returnValue()), add=TRUE)
  # ... the rest of Foo
}
```

The magic R function `returnValue()` in an `on.exit()` will contain
the value that the function returns, no matter how it returns it.

R has an alternative trace() facility, but it only displays the arguments 
to a function in unevaluated form, so it's very difficult to see what 
parameters the caller actually passed to a given function call.

### CODE REVIEW

File: R/internal.R [hunk 1]
```diff
@@ -119,7 +134,7 @@ tmpnam = function(db, prefix="R_array")
 {
   stopifnot(inherits(db, "afl"))
   salt = basename(tempfile(pattern=prefix))
-  paste(salt, getuid(db), sep="")
+  paste(salt, getuid(db), sep="_")
```
Comment (rvernica) [inline R/internal.R:137]: Is this extra space on purpose?

**Review by rvernica — CHANGES_REQUESTED** [review]
Looks good. I would add the blob that you have in the PR comment about the different options to the `README` so we can quickly find it in the future.

File: R/internal.R (hunk 1 above)
Comment (rls-p4) [reply R/internal.R:137]: Yes, it helped me to visually distinguish the connection ID from the temporary array name, which was helpful when I needed to clean up leftover arrays from old connections (which is fixed now). I can revert the change but I strongly prefer it this way.

**Comment by rls-p4** [comment]
> Looks good. I would add the blob that you have in the PR comment about the different options to the `README` so we can quickly find it in the future.

Added a section on debug logging to README.md.

**Review by rvernica — APPROVED** [review]

**Pull request merged**